src/js/components/List/List.js

```diff
@@ -277,16 +277,40 @@ const List = React.forwardRef(
                 let boxProps = {};
                 let itemId;
 
-                if (children) {
+                if (children && primaryKey) {
+                  content = children(
+                    item,
+                    index,
+                    onClickItem ? { active: active === index } : undefined,
+                  );
+                  itemId = normalize(item, index, primaryKey);
+                } else if (children) {
                   content = children(
                     item,
                     index,
                     onClickItem ? { active: active === index } : undefined,
                   );
                 } else if (primaryKey) {
                   if (typeof primaryKey === 'function') {
-                    itemId = primaryKey(item, index);
-                    content = itemId;
+                    content = primaryKey(item, index);
+                    if (!content?.key)
+                      console.error(
+                        `You must add key to your element in primaryKey func.`,
+                      );
```

Comment:
I think in the case where there is a primaryKey function but the user doesn't specify a key in the function it shouldn't result in an error and instead the key should default to index values from 0 to n